rpc/namespaces/eth/api.go

```diff
@@ -571,14 +650,71 @@ func (api *PublicEthereumAPI) GetBlockByHash(hash common.Hash, fullTx bool) (map
 // GetBlockByNumber returns the block identified by number.
 func (api *PublicEthereumAPI) GetBlockByNumber(blockNum rpctypes.BlockNumber, fullTx bool) (map[string]interface{}, error) {
 	api.logger.Debug("eth_getBlockByNumber", "number", blockNum, "full", fullTx)
-	return api.backend.GetBlockByNumber(blockNum, fullTx)
+
+	if blockNum != rpctypes.PendingBlockNumber {
+		return api.backend.GetBlockByNumber(blockNum, fullTx)
+	}
+
+	height, err := api.backend.LatestBlockNumber()
+	if err != nil {
+		return nil, err
+	}
+
+	// latest block info
+	latestBlock, err := api.clientCtx.Client.Block(&height)
+	if err != nil {
+		return nil, err
+	}
+
+	// number of pending txs queried from the mempool
+	unconfirmedTxs, err := api.clientCtx.Client.UnconfirmedTxs(1000)
```

Comment:
What if the actual tx quantity is greater than 1000?

Reply:
It's highly unlikely (for now) as on the Cosmos Hub there are 1 or 2 avg txs per block.